pyomo/repn/plugins/gams_writer.py

```diff
@@ -252,13 +252,21 @@ def _write_model(self,
         for con in model.component_data_objects(Constraint,
                                                 active=True,
                                                 sort=sort):
-            if skip_trivial_constraints and con.body.is_fixed():
+
+            if (not con.has_lb()) and \
+               (not con.has_ub()):
+                assert not con.equality
+                continue # non-binding, so skip
```

Comment:
There is such thing as a non-binding constraint? What purpose do those serve? And how does one generate such a thing because this didn't work:
```python
>>> m.x = Var()
>>> m.c = Constraint(expr=m.x * 2 + 10)
ERROR: Constructing component 'c' from data=None failed:
        ValueError: Constraint 'c' does not have a proper value. Found '10 + 2*x'
        Expecting a tuple or equation. Examples:
           summation(model.costs) == model.income
           (0, model.price[item], 50)
```